File: solutions_2692487_0/Python/Fedux/motes.py
#!/usr/bin/env python

import sys
import operator
from math import sqrt, floor, ceil
from collections import Counter

class InputFile:
	def __init__(self, fd):
		self.fd = fd
	def readInt(self):
		return int(self.fd.readline())
	def readIntegers(self):
		return tuple([int(x) for x in self.fd.readline().split()])
	def readIntegersList(self):
		return [int(x) for x in self.fd.readline().split()]
	def readString(self):
		return self.fd.readline()[:-1]

def solve(a, motes):
	m.sort()
	azioni_fatte = 0
	azioni_per_finire = []
	
	for i in range(len(motes)):
		s = motes[i]
		mancanti = len(motes)-i
		
		if s >= a:
			#print "Sono ", a, "e non riesco a mangiare", s
			azioni_per_finire.append( azioni_fatte + mancanti )
			
			while s >= a:
				#print " + Sono ", a, "e aumento a", (a+a-1)
				azioni_fatte += 1
				a += a-1
		a += s
		
	azioni_per_finire.append( azioni_fatte )
	
	return min(azioni_per_finire)


inputfile = InputFile(sys.stdin)
T = inputfile.readInt()
for case in range(1,T+1):
	(a, n) = inputfile.readIntegers()
	m = inputfile.readIntegersList()
	
	if n != len(m):
		print "ERRORE"
		sys.exit(1)
	
	if a == 1:
		result = len(m)
	else:
		result = solve(a, m)
	
	print "Case #%d: %d" % (case, result)

File: solutions_2692487_0/C++/Salmon/motes.cpp
#include <cstdio>
#include <algorithm>
using namespace std;
int mote[1000010];
void solve()
{
    int a,n,mi=10000000,ad=0;
    scanf("%d%d",&a,&n);
    for(int i=0;i<n;i++)
        scanf("%d",&mote[i]);
    if(a==1)
    {
        printf("%d",n);
        return;
    }
    sort(mote,mote+n);
    for(int i=0;i<n;i++)
    {
        mi=min(mi,ad+n-i);
        if(a<=mote[i])
        {
            while(a<=mote[i])
            {
                ad++;
                a+=a-1;
            }
        }
        a+=mote[i];
    }
    mi=min(mi,ad);
    printf("%d",mi);
}
int main()
{
    int t;
    freopen("Sm.in","r",stdin);
    freopen("ou.out","w",stdout);
    scanf("%d",&t);
    for(int i=1;i<=t;i++)
    {
        printf("Case #%d: ",i);
        solve();
        printf("\n");
    }
}
